### PR TITLE
Add Google Cloud Storage support to logshare

### PR DESCRIPTION
**Pull request by garrettgalow**
Added two params for specifying project and bucket to upload logshare data to GCS.

### CODE REVIEW

**Comment by elithrar** [comment]
@garrettgalow Run gofmt over your commits - https://travis-ci.org/cloudflare/logshare/jobs/273119833#L614-L769

File: cmd/logshare-cli/main.go [hunk 1]
```diff
@@ -4,11 +4,16 @@ import (
 	"log"
 	"os"
 	"time"
+	s "strings"
```
Comment (elithrar) [inline cmd/logshare-cli/main.go]: No need to alias this, `strings` is the convention.

File: cmd/logshare-cli/main.go [hunk 2]
```diff
@@ -4,11 +4,16 @@ import (
...
+	"strconv"
 
 	cloudflare "github.com/cloudflare/cloudflare-go"
 	"github.com/cloudflare/logshare"
 	"github.com/pkg/errors"
 	"github.com/urfave/cli"
+	//packages for writing to google cloud storage
```
Comment (elithrar) [inline cmd/logshare-cli/main.go]: No need for this comment. Let goimports format this.

File: cmd/logshare-cli/main.go [hunk 3]
```diff
@@ -4,11 +4,16 @@ import (
...
+        gStr "cloud.google.com/go/storage"
```
Comment (elithrar) [inline cmd/logshare-cli/main.go]: Leave as `storage`, or perhaps `gcs`.

File: cmd/logshare-cli/main.go [hunk 4]
```diff
@@ -137,6 +182,11 @@ func (conf *config) Validate() error {
 	// 	return errors.New("count must be > 0, or set to -1 (no limit)")
 	// }
 
+	if (conf.googleStorageBucket != "" && conf.googleProjectId == "") ||
```
Comment (elithrar) [inline cmd/logshare-cli/main.go]: Clarify? If both are needed, just `if conf.googleStorageBucket == "" || conf.googleProjectId == ""` is sufficient.

**Comment by garrettgalow** [comment]
Ok addressed your comments, and fixed an issue I found. Tested writing to stdout, writing to gcs, and when providing only 1 of each gcs parameter.

File: cmd/logshare-cli/main.go (hunk 3 above)
Comment (garrettgalow) [reply cmd/logshare-cli/main.go]: changed this to gcs

**Comment by garrettgalow** [comment]
my guess is the travis build fails since the google go library only supports 1.6+
https://github.com/GoogleCloudPlatform/google-cloud-go/blob/master/.travis.yml

**Comment by elithrar** [comment]
We can probably drop 1.6 now that 1.9 is out. Want to update the Travis
config?

**Comment by garrettgalow** [comment]
done

**Pull request merged**